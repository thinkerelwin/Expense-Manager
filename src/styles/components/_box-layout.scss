.box-layout {
  display: flex;
  height: 100vh;
  width: 100vw;
  justify-content: center;
  align-items: center;
  background: url('/images/bg.jpg');
  background-size: cover;

}

.box-layout__box {
  width: 28rem;
  padding: $l-size $m-size;
  text-align: center;
  background: fade-out($color: white, $amount: .15);
  border-radius: 5px;

}

.box-layout__title {
  margin: 0 0 $m-size 0;
  line-height: 1;

}