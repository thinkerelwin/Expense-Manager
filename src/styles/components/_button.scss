.button {
  display: inline-block;
  padding: $s-size;
  border: none;
  text-decoration: none;
  line-height: 1;
  font-size: $font-size-large;
  font-weight: 300;
  color: white;
  background-color: $blue;
}

.button--secondary {
  background-color: #888888;
}

.button--link {
  background: none;
}