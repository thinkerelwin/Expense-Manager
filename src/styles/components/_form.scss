.form {
  display: flex;
  flex-direction: column;
  > * {
    margin-bottom: $m-size;
  }
}

.form__error {
  margin-right: $m-size;
  font-style: italic; 
}