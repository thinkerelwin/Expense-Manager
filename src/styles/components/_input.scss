.text-input {
  height: 50px;
  padding: $s-size;
  font-size: $font-size-large;
  font-weight: 300;
  border: 1px solid #cacccd;
}

.select {
  @extend .text-input;
}

.textarea {
  @extend .text-input;
  height: 10rem;
}