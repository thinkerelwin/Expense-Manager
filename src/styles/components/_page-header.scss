.page-header {
  padding: $l-size 0;
  margin-bottom: $l-size;
  background: $off-white;

}

.page-header__title {
  margin: 0;
  font-weight: 300;
  span {
    font-weight: bold;
  }
}

.page-header__actions {
  margin-top: $m-size;
}