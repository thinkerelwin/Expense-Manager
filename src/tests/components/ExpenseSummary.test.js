import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'

test('should correctly render ExpenseSummary with 1 expense ', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={227}/>)
  expect(wrapper).toMatchSnapshot()
})

test('should correctly render ExpenseSummary with multiple expense ', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={3} expensesTotal={722} />)
  expect(wrapper).toMatchSnapshot()
})