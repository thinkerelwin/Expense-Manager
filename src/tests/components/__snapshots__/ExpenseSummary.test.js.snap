// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpenseSummary with 1 expense  1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $2.27
  </h1>
</div>
`;

exports[`should correctly render ExpenseSummary with multiple expense  1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $7.22
  </h1>
</div>
`;
